pkg/kv/kvserver/replica_rangefeed_test.go

```diff
@@ -779,6 +780,57 @@ func TestReplicaRangefeedRetryErrors(t *testing.T) {
 		pErr := <-streamErrC
 		assertRangefeedRetryErr(t, pErr, roachpb.RangeFeedRetryError_REASON_LOGICAL_OPS_MISSING)
 	})
+	t.Run(roachpb.RangeFeedRetryError_REASON_ADDSSTABLE.String(), func(t *testing.T) {
+		tc, rangeID := setup(t)
+		defer tc.Stopper().Stop(ctx)
+
+		// Establish a rangefeed on the replica we ingest into.
+		stream := newTestStream()
+		streamErrC := make(chan *roachpb.Error, 1)
+		rangefeedSpan := roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("z")}
+		ts := tc.Servers[0]
+		store, err := ts.Stores().GetStore(ts.GetFirstStoreID())
+		require.NoError(t, err)
+		go func() {
+			req := roachpb.RangeFeedRequest{
+				Header: roachpb.Header{
+					RangeID: rangeID,
+				},
+				Span: rangefeedSpan,
+			}
+			streamErrC <- store.RangeFeed(&req, stream)
+		}()
+
+		// Wait for the first checkpoint event.
+		waitForInitialCheckpointAcrossSpan(t, stream, streamErrC, rangefeedSpan)
+
+		// Ingest an SST.
+		sstFile := &storage.MemFile{}
+		writer := storage.MakeBackupSSTWriter(sstFile)
+		defer writer.Close()
+
+		require.NoError(t, writer.PutMVCC(storage.MVCCKey{
+			Key:       roachpb.Key("foo"),
+			Timestamp: hlc.Timestamp{WallTime: 1},
+		}, roachpb.MakeValueFromString("bar").RawBytes))
+		require.NoError(t, writer.Finish())
+		writer.Close()
+
+		req := &roachpb.AddSSTableRequest{
+			RequestHeader: roachpb.RequestHeader{
+				Key:    roachpb.Key("f"),
+				EndKey: roachpb.Key("g"),
+			},
+			Data:                    sstFile.Bytes(),
+			WriteAtRequestTimestamp: true,
+		}
+		_, pErr := kv.SendWrapped(ctx, store.TestSender(), req)
+		require.Nil(t, pErr)
+
+		// Check the error.
+		pErr = <-streamErrC
```

Comment:
Wouldn't that make this test hang if error is not generated? Other tests seem to use
```
			timer := time.AfterFunc(10*time.Second, stream.Cancel)
			defer timer.Stop()
```
inside the feed go routine right before creating a feed to avoid waiting indefinitely.

Reply:
Not really. We're populating that channel right above, in a separate goroutine:

https://github.com/cockroachdb/cockroach/blob/2428b09aad2a73e08249efe3a8b8cc8e2a19c8ee/pkg/kv/kvserver/replica_rangefeed_test.go#L794-L802

So the channel will yield the return value of `RangeFeed()`, regardless of whether that's an error or `nil`. It will only hang if that goroutine panics before `RangeFeed()` returns a value, which seems fine.

Reply:
No, I think you're right -- seemed to remember that `RangeFeed()` returned immediately, but of course it doesn't. Fixed.